engine/configurator/conf_cyaml.c

```diff
@@ -0,0 +1,796 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "conf_cyaml.h"
+
+/******************************************************************************
+ * CYAML schema to tell libcyaml about both expected YAML and data structure.
+ *
+ * (Our CYAML schema is just a bunch of static const data.)
+ ******************************************************************************/
+
+/* Mapping from instance or object strings to enum values for schema. */
+const cyaml_strval_t scope_strings[] = {
+    {"object",   CFG_OBJECT},
+    {"instance", CFG_INSTANCE},
+};
+
+/* Mapping from 'access' strings to enum values for schema. */
+const cyaml_strval_t access_strings[] = {
+    {"read_write",  CFG_READ_WRITE },
+    {"read_only",   CFG_READ_ONLY  },
+    {"read_create", CFG_READ_CREATE},
+};
+
+/* Mapping from 'type' strings to flag enum for schema. */
+const cyaml_strval_t type_strings[] = {
+    {"none",    CVT_NONE},
+    {"bool",    CVT_BOOL},
+    {"int8",    CVT_INT8},
+    {"uint8",   CVT_UINT8},
+    {"int16",   CVT_INT16},
+    {"uint16",  CVT_UINT16},
+    {"int32",   CVT_INT32},
+    {"integer", CVT_INT32},
+    {"uint32",  CVT_UINT32},
+    {"int64",   CVT_INT64},
+    {"uint64",  CVT_UINT64},
+    {"string",  CVT_STRING},
+    {"address", CVT_ADDRESS},
+};
+
+/* Mapping from no_parent_dep strings to enum values for schema. */
+const cyaml_strval_t no_parent_dep_strings[] = {
+    {"yes", FALSE}, /* it is a default value */
+    {"no",  TRUE},
+};
+
+/* The depends mapping's field definitions schema is an array.
```

Comment:
Multi-line comments must have empty first line:
```
/*
  * The depends mapping's field definitions schema is an array.
  ...
  */
```

Also I suggest using Doxygen-style comments for such extensive documentation.

Reply:
I've switched the comments to Doxygen-style.